api/grpcserver/v2alpha1/malfeasance.go

```diff
@@ -176,68 +230,19 @@ func (s *MalfeasanceStreamService) Stream(
 				}
 			case <-eventsFull:
 				return status.Error(codes.Canceled, "buffer overflow")
-			case rst, ok := <-dbChan:
-				if !ok {
-					dbChan = nil
-					if sub == nil {
-						return nil
-					}
-					continue
-				}
-				err = stream.Send(rst)
-				switch {
-				case errors.Is(err, io.EOF):
-					return nil
-				case err != nil:
-					return status.Error(codes.Internal, err.Error())
-				}
-			case err := <-errChan:
-				return err
 			case <-stream.Context().Done():
 				return nil
 			}
 		}
 	}
 }
 
-func (s *MalfeasanceStreamService) fetchFromDB(
-	ctx context.Context,
-	ops builder.Operations,
-) (<-chan *spacemeshv2alpha1.MalfeasanceProof, <-chan error) {
-	dbChan := make(chan *spacemeshv2alpha1.MalfeasanceProof)
-	errChan := make(chan error, 1) // buffered to avoid blocking, routine should exit immediately after sending an error
-
-	go func() {
-		defer close(dbChan)
-		if err := identities.IterateOps(s.db, ops,
-			func(id types.NodeID, proof []byte, received time.Time) bool {
-				rst := toProof(ctx, s.info, id, proof)
-				if rst == nil {
-					return true
-				}
-
-				select {
-				case dbChan <- rst:
-					return true
-				case <-ctx.Done():
-					// exit if the context is canceled
-					return false
-				}
-			},
-		); err != nil {
-			errChan <- status.Error(codes.Internal, err.Error())
-		}
-	}()
-	return dbChan, errChan
-}
-
 func toProof(
```

Comment:
I actually don't quite like the naming here. `toSmth` usually implies a simple operation that converts between different representations. It used to be this way, but now it involves a database query. So maybe it should be named `loadProof` or smth like that?

Reply:
Updated to `fetcMetaData`